hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/MultipartInputStream.java

```diff
@@ -173,6 +178,28 @@ public synchronized void seek(long pos) throws IOException {
     prevPartIndex = partIndex;
   }
 
+  public synchronized void initialize() throws IOException {
```

Comment:
@jojochuang , the problem we are now facing is how to make sure a reader of an open file will succeed. There is two things left last time when I investigated this issue, 
a. how to handle if the writer of the open file is slower than the reader of the open file. Say block A, it's block size in OM is 10 bytes, when the reader starts, the reader fetches the block A length from DN which is 80 bytes,  and later,  the writer writes more data, then length becomes 90 bytes, should the reader refetch the block size from DN again in this case? 
b. what if there is a new block allocated by the writer?

Reply:
For the HBase replication case, the application expects the input stream to read until the hsync length returned by output stream before the input stream instantiates. So for your case the read shouldn't read beyond 80 bytes. And therefore, new blocks added by the writer shouldn't matter either.

Reply:
BTW I have applied the fix to the HBase cluster. It looks like it's working with no errors. I'm still in the process of verifying that data does get written to the destination cluster though.

Reply:
If HBase reader needs read to the length of output stream when reader is instantiated, then we don't have to handle the two cases mentioned above.  And this patch looks good to me. Thanks @jojochuang .